Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -1000,6 +1001,17 @@ bool HTMLInputElement::isTextType() const
     return m_inputType->isTextType();
 }
 
+void HTMLInputElement::setDefaultCheckedState(bool isDefaultChecked)
+{
+    if (m_isDefaultChecked == isDefaultChecked)
+        return;
+
+    std::optional<Style::PseudoClassChangeInvalidation> defaultInvalidation;
+    if (m_inputType->isCheckable())
+        defaultInvalidation.emplace(*this, CSSSelector::PseudoClassType::Default, isDefaultChecked);
+    m_isDefaultChecked = isDefaultChecked;
+}
```

Comment:
Can we add a wpt for this case involving :has()? Expanding the existing one should be fine.

Reply:
https://github.com/web-platform-tests/wpt/pull/41903